### PR TITLE
JENKINS-42185 : Basic pipeline slicing support

### PR DESCRIPTION
**Pull request by mslusarczyk**
Changes needed to support pipeline.
- https://github.com/jenkinsci/configurationslicing-plugin/compare/master...mslusarczyk:pipeline_support?expand=1#diff-323784f136473ceb742328d3b9ea2347R28 extended to support more than AbstractProject
- Trigger slicers changed to support WorkflowJob
- Concurrent builds slicer added
- Pipeline script slicer added

### CODE REVIEW

**Comment by duemir** [comment]
https://issues.jenkins-ci.org/browse/JENKINS-42185

**Comment by raphtheb** [comment]
For the record, and posterity, this PR is gold. I built the plugin from it and it worked flawlessly. I can't do much more than +1 it. 

Apologies for pinging something from way back in February!

**Comment by mslusarczyk** [comment]
@raphtheb You can find few more things in my fork, if you're interested.

**Comment by guysoft** [comment]
@mslusarczyk Hey, can you link to your fork? Which seems to be not there anymore?

Also, if you have a build of this that can be installed that would be great!

**Comment by oleg-nenashev** [comment]
CC @jenkinsci/code-reviewers . Feedback will be appreciated so that we can assist the new maintainer to get it over the line (see https://groups.google.com/forum/#!topic/jenkinsci-dev/O2cO3zXuDi0)

**Comment by guysoft** [comment]
Hey, so I built and test this here and it seems to work, I am new to this org and it would be nice to get some confirmation that this is a good commit before merging it.

File: src/main/java/configurationslicing/AbstractJob.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            ((WorkflowJob) item).setConcurrentBuild(value);
+
+        throw new IOException("Unsupported job type");
+    }
+
+    public <T extends Trigger> T getTrigger(Class<T> clazz) {
+        if (item instanceof AbstractProject)
+            return (T) ((AbstractProject) item).getTrigger(clazz);
+        if (item instanceof WorkflowJob) {
+            return getTrigger((WorkflowJob) item, clazz);
```
Comment (jglick) [inline src/main/java/configurationslicing/AbstractJob.java:51]: Some of these things are better handled via `ParameterizedJob` by the way.

Comment (jglick) [reply src/main/java/configurationslicing/AbstractJob.java:51]: or `SCMTriggerItem`

Comment (guysoft) [reply src/main/java/configurationslicing/AbstractJob.java:51]: @jglick  
Hey new and kinda trying to make sense of the codebase and Jenkins Core API.

Can you be more specific? What would ``SCMTriggerItem`` and ``ParameterizedJob`` replace?

Comment (jglick) [reply src/main/java/configurationslicing/AbstractJob.java:51]: Explicit checks for `AbstractProject` and `WorkflowJob`. Not sure all functionality you are looking for here would be covered.

**Pull request merged**